### PR TITLE
Feature/rs basic infra

### PR DESCRIPTION
**Pull request by rayshih**
This is an unfinished feature. However, it is an almost finished infrastructure. This PR is created for aggregating some feedbacks.

The followings are what I'm going to do next:

1. create web interfaces to setup slack token and config slack channel
2. port some examples from https://github.com/botpress/botpress-examples
    - maybe extend those modules to support both connectors.
3. handle all event from slack

### CODE REVIEW

File: src/slackConnector.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  const authenticateP = new Promise((resolve) => {
+    rtm.on(CLIENT_EVENTS.RTM.AUTHENTICATED, (rtmStartData) => {
+      data = rtmStartData
+      resolve(data)
+    })
+  })
+
+  rtm.on(CLIENT_EVENTS.RTM.RTM_CONNECTION_OPENED, () => {
+    isConnected = true
```
Comment (slvnperron) [inline src/slackConnector.js]: We also need to handle the disconnection event
https://github.com/slackapi/node-slack-sdk/blob/cbe9da5fbd38a22cbf7fe7b7253a6fdefe6a4df3/lib/clients/events/client.js#L26

File: src/slackConnector.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// - connection failed
+//
+export default (slackApiToken, sendIncoming) => {
+  let data
+  let isConnected = false
+
+  const rtm = new RtmClient(slackApiToken)
+
+  const authenticateP = new Promise((resolve) => {
+    rtm.on(CLIENT_EVENTS.RTM.AUTHENTICATED, (rtmStartData) => {
```
Comment (slvnperron) [inline src/slackConnector.js]: Reject the promise on auth error:

https://github.com/slackapi/node-slack-sdk/blob/cbe9da5fbd38a22cbf7fe7b7253a6fdefe6a4df3/lib/clients/events/client.js#L35

File: src/slackConnector.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import Promise from 'bluebird'
+import { RtmClient, CLIENT_EVENTS, RTM_EVENTS } from '@slack/client'
+
+// TODO handle all events from slack
+//
+// - connection failed
+//
```
Comment (slvnperron) [inline src/slackConnector.js]: Add the following events to the TODO:
- Rate limited
- Disconnected

File: src/slackConnector.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+export default (slackApiToken, sendIncoming) => {
+  let data
+  let isConnected = false
+
+  const rtm = new RtmClient(slackApiToken)
+
+  const authenticateP = new Promise((resolve) => {
```
Comment (slvnperron) [inline src/slackConnector.js]: We probably want to timeout the login after 30 seconds

File: src/index.js [hunk 5]
```diff
@@ -1,8 +1,37 @@
...
+    bp.slack = createSendFuncs(bp.middlewares.sendOutgoing)
+
+    const slackApiToken = config.slackApiToken.get()
+    // TODO check token
+    slackConn = SlackConnector(slackApiToken, bp.middlewares.sendIncoming)
+
+    // TODO channel list api
+    // TODO select channel api
+    slackConn.authenticateP.done(data => {
+      channel = data.channels.filter(c => c.name === channelName)[0]
```
Comment (slvnperron) [inline src/index.js]: handle the case where channel isn't found

**Comment by rayshih** [comment]
@slvnperron updated!

**Pull request merged**